pkg/templates/pod.tmpl

```diff
@@ -15,7 +15,18 @@ metadata:
     "{{ $key }}": "{{ $val }}"
 {{ end }}
 spec:
-  nodeName: "{{.NodeName}}"
+  affinity:
+    podAntiAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+      - labelSelector:
+          matchExpressions:
+{{ range $key, $val := .Labels }}
+          - key: {{ $key }}
+            operator: In
+            values:
+            - {{ $val }}
```

Comment:
What if 2 CRs have the same labels? Should we include the kvc name here instead?